e2e/cli-plugins/flags_test.go

```diff
@@ -17,3 +19,82 @@ func TestRunGoodArgument(t *testing.T) {
 		Out:      "Hello Cleveland!",
 	})
 }
+
+// TestClashWithGlobalArgs ensures correct behaviour when a plugin
+// has an argument with the same name as one of the globals.
+func TestClashWithGlobalArgs(t *testing.T) {
```

Comment:
(not a blocker for this PR); we should evaluate the naming of these tests, so make them easier to "grep" (which can be useful to run a subset of tests), i.e. give them a consistent prefix (e.g. `TestPluginGlobalArgsClash`, `TestPluginGlobalArgsUnknown`)